plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryFilterQueryBuilder.java

```diff
@@ -72,36 +72,50 @@ private String concat(List<String> clauses)
         for (BigQueryColumnHandle column : columns) {
             Domain domain = tupleDomain.getDomains().get().get(column);
             if (domain != null) {
-                clauses.add(toPredicate(column.getName(), domain, column));
+                toPredicate(column.getName(), domain, column).ifPresent(clauses::add);
             }
         }
         return clauses.build();
     }
 
-    private String toPredicate(String columnName, Domain domain, BigQueryColumnHandle column)
+    private Optional<String> toPredicate(String columnName, Domain domain, BigQueryColumnHandle column)
     {
         if (domain.getValues().isNone()) {
-            return domain.isNullAllowed() ? quote(columnName) + " IS NULL" : "FALSE";
+            String predicate = domain.isNullAllowed() ? quote(columnName) + " IS NULL" : "FALSE";
+            return Optional.of(predicate);
         }
 
         if (domain.getValues().isAll()) {
-            return domain.isNullAllowed() ? "TRUE" : quote(columnName) + " IS NOT NULL";
+            String predicate = domain.isNullAllowed() ? "TRUE" : quote(columnName) + " IS NOT NULL";
+            return Optional.of(predicate);
         }
 
         List<String> disjuncts = new ArrayList<>();
         List<Object> singleValues = new ArrayList<>();
         for (Range range : domain.getValues().getRanges().getOrderedRanges()) {
             checkState(!range.isAll()); // Already checked
             if (range.isSingleValue()) {
+                Optional<String> value = column.getBigQueryType().convertToString(column.getTrinoType(), range.getSingleValue());
```

Comment:
`convertToString` is done and thrown away.

while we could have `canConvertToString` (which under the hood would do what we do here), 
it would be more in "the spirit of Optional" to call `convertToString`  once and
use the result for both a) returning early if empty and b) rendering the final SQL

for this, you would need to make `singleValues` into `List<String>`

Reply:
@findepi Changed `singleValues` to `List<String>` in [c06b146205343f3a22bafd3fdb4791ddb6c6c483..2a63eda48ad011c03f9b2d0ba31e5b744af8e000](https://github.com/trinodb/trino/compare/c06b146205343f3a22bafd3fdb4791ddb6c6c483..2a63eda48ad011c03f9b2d0ba31e5b744af8e000) Please let me know if we need additional test case for this change.